iroh-sync/src/sync.rs

```diff
@@ -1943,56 +1943,167 @@ mod tests {
+                Query::single_latest_per_key().key_prefix("hi/")
+            ),
+            expect(&[("hi/moon", &a2), ("hi/world", &a1)]),
         );
+
         assert_eq!(
-            res,
-            vec![
-                ("hello/world".to_string(), a2.id()),
-                ("hello/world".to_string(), a1.id()),
-                ("hello/moon".to_string(), a1.id()),
-                ("hello".to_string(), a3.id()),
-            ]
+            query(
+                &store,
+                namespace,
+                Query::single_latest_per_key()
+                    .key_prefix("hi/")
+                    .sort_direction(SortDirection::Desc)
+            ),
+            expect(&[("hi/world", &a1), ("hi/moon", &a2)]),
+        );
+
+        assert_eq!(
+            query(
+                &store,
+                namespace,
+                Query::all().sort_by(SortBy::KeyAuthor, SortDirection::Asc)
+            ),
+            expect(&[
+                ("hi", &a3),
+                ("hi/moon", &a2),
+                ("hi/world", &a1),
+                ("hi/world", &a2),
+            ]),
+        );
+
+        assert_eq!(
+            query(
+                &store,
+                namespace,
+                Query::all().sort_by(SortBy::KeyAuthor, SortDirection::Desc)
+            ),
+            expect(&[
+                ("hi/world", &a2),
+                ("hi/world", &a1),
+                ("hi/moon", &a2),
+                ("hi", &a3),
+            ]),
+        );
+
+        assert_eq!(
+            query(&store, namespace, Query::all().key_prefix("hi/")),
+            expect(&[("hi/world", &a1), ("hi/moon", &a2), ("hi/world", &a2),]),
+        );
+
+        assert_eq!(
+            query(
+                &store,
+                namespace,
+                Query::all()
+                    .key_prefix("hi/")
+                    .sort_by(SortBy::KeyAuthor, SortDirection::Desc)
+            ),
+            expect(&[("hi/world", &a2), ("hi/world", &a1), ("hi/moon", &a2),]),
+        );
+
+        assert_eq!(
+            query(
+                &store,
+                namespace,
+                Query::all()
+                    .key_prefix("hi/")
+                    .sort_by(SortBy::AuthorKey, SortDirection::Asc)
+            ),
+            expect(&[("hi/world", &a1), ("hi/moon", &a2), ("hi/world", &a2),]),
+        );
+
+        assert_eq!(
+            query(
+                &store,
+                namespace,
+                Query::all()
+                    .key_prefix("hi/")
+                    .sort_by(SortBy::AuthorKey, SortDirection::Desc)
+            ),
+            expect(&[("hi/world", &a2), ("hi/moon", &a2), ("hi/world", &a1)]),
+        );
+
+        assert_eq!(
+            query(
+                &store,
+                namespace,
+                Query::all()
+                    .sort_by(SortBy::KeyAuthor, SortDirection::Asc)
+                    .limit(2)
+                    .offset(1)
```

Comment:
I think there is no test with key_prefix mixed with offset/limit, that would be good to have

Reply:
Added a test for that